### PR TITLE
MongoDB maxPoolSize to env variable. Refs: #3425

### LINKED ISSUE #3425: [NV-2341] 🐛 Bug Report: MongoDB huge amount of connections
### 📜 Description

Ive deployed Novu to a Google Cloud Cluster with a MongoDB Deployment (using MongoDB Atlas in Google Cloud)

it is just a plain install with the API, WEB, WORKER and WS container. 

pretty instant after de deployment I receive messages from MongoDB that the connection limit is reached. (500) 

wondering why the application is using almost 500 mongoDB connections at the same time while doing basicly nothing. 

### 👟 Reproduction steps

Deploy on Google Cloud, with MongoDB (using MongoDB Atlas)

### 👍 Expected behavior

not that many database connection. (solution in [Discord](https://discord.com/channels/895029566685462578/1107967383332732989) was that connections should be configured by an ENV)

### 👎 Actual Behavior with Screenshots

image

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes

<sub>[NV-2341](https://linear.app/novu/issue/NV-2341/🐛-bug-report-mongodb-huge-amount-of-connections)</sub>

### PR DESCRIPTION
**Pull request by wouter-toppy**
### What change does this PR introduce?
MongoDB maxPoolSize to an env variable

### Why was this change needed?

Closes #3425

### CODE REVIEW

File: apps/api/src/config/env-validator.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ const validators: { [K in keyof any]: ValidatorSpec<any[K]> } = {
     default: '',
   }),
   MONGO_URL: str(),
+  MONGO_MAX_POOL_SIZE: num({
+    default: 700,
```
Comment (p-fernandez) [inline apps/api/src/config/env-validator.ts:35]: Considering you are being warn in MongoDB Atlas that is over 500, maybe that should be the default limit. Same, for the .env files.
For Novu's own configuration we can set that feature flag once this is merged.

**Review by p-fernandez — CHANGES_REQUESTED** [review]
Thanks for your contribution. 
🌟

**Comment by wouter-toppy** [comment]
@scopsy  done!

**Review by p-fernandez — APPROVED** [review]
🌟

**Comment by wouter-toppy** [comment]
@p-fernandez  pipeline is still failing, what can we do about it? already did a rebase.

**Comment by p-fernandez** [comment]
> @p-fernandez pipeline is still failing, what can we do about it? already did a rebase.

That error is something on our side. No worries. Thanks for pinging me.

**Pull request merged**